### ISSUE TITLE
Locations for mssql-jdbc_auth-xxx.dll restricted to java.library.path

### ISSUE THREAD

**Issue by mlenz968** [microsoft/mssql-jdbc#1396]
## Driver version
8.2.2.jre8

## Client Operating System
Windows

## Problem description
In older versions it was possible to put the sqljdbc_auth.dll to a folder lib\win-x86_64 besides the mssql-jdbc-xxx.jar
Now it seems that the dll can only be loaded from the java.library.path.
1. Expected behaviour:
    mssql-jdbc_auth-xxx.dll can be loaded from the same folder where mssql-jdbc-xxx.jar is located.
2. Actual behaviour:
    mssql-jdbc_auth-xxx.dll can only be loaded from java.library.path
3. Error message/stack trace:
    java.lang.UnsatisfiedLinkError: no mssql-jdbc_auth-8.2.2.x64 in java.library.path: [...]

## Reproduction code
`public class JdbcConnectionTest
{
    public static void main(String[] args) throws SQLException
    {
        try (Connection connection = DriverManager.getConnection(args[0]))
        {
            System.out.println(connection);
        }
    }
}`

1. Create following folder structure:
    test
    test\classes
    test\lib
2. Put JdbcConnectionTest.class to test\classes
3. Put mssql-jdbc-8.2.2.jre8.jar to test\lib
4. Put mssql-jdbc_auth-8.2.2.x64.dll to test\lib
5. From working directory test execute the following command line:
    java -cp classes;lib\mssql-jdbc-8.2.2.jre8.jar JdbcConnectionTest jdbc:sqlserver://<connection_string>;integratedSecurity=true
6. Test will fail with java.lang.UnsatisfiedLinkError: no mssql-jdbc_auth-8.2.2.x64 in java.library.path: [...]
7. Put mssql-jdbc_auth-8.2.2.x64.dll directly to the working directory test
8. Try again and test will succeed

**Comment by ulvii** [comment]
Hi @mlenz968 ,
The new drivers use the same `System.loadLibrary()` method which loads the DLL from `java.library.path` as before. I am not sure why you are seeing the difference, perhaps there are changes in your environment. You need to make sure test\lib folder is in `java.library.path`.

**Comment by mlenz968** [comment]
Hi @ulvii ,
Sorry I suspected the problem in the new jdbc driver but my expected behavior was not implemented in the old driver also.
It was implemented in the Wildfly/JBoss module loader (see [JBoss Modules - Native Libraries](https://jboss-modules.github.io/jboss-modules/manual/#native-libraries)).
I also upgraded Wildfly from 16.0 to 20.0 and it seems that the new version has some changes/bug? in the module loader.
I will investigate into this direction now.
